cirq/pasqal/pasqal_qubits.py

```diff
@@ -11,184 +11,199 @@
 
     @staticmethod
     def parallelep(rows: int,
                    cols: int,
                    lays: int,
-                   top: int = 0,
-                   left: int = 0,
-                   upper: int = 0) -> List['ThreeDGridQubit']:
-        """Returns a parallelepiped of ThreeDGridQubits.
+                   x0: float = 0,
+                   y0: float = 0,
+                   z0: float = 0) -> List['ThreeDQubit']:
+        """Returns a parallelepiped of ThreeDQubits.
 
         Args:
             rows: Number of rows in the rectangle
             cols: Number of columns in the rectangle
-            top: Row number of the topmost row
-            left: Column number of the leftmost row
+            x0: x-coordinate of the first qubit
+            y0: y-coordinate of the first qubit
+            z0: z-coordinate of the first qubit
 
         Returns:
-            A list of ThreeDGridQubits filling in a rectangular grid
+            A list of ThreeDQubits filling in a 3D grid
         """
         return [
-            ThreeDGridQubit(row, col, lay) for row in range(top, top + rows)
-            for col in range(left, left + cols)
-            for lay in range(upper, upper + lays)
+            ThreeDQubit(x0 + x, y0 + y, z0 + z) for z in range(lays)
+            for y in range(cols) for x in range(rows)
         ]
 
+    def __repr__(self):
+        return 'pasqal.ThreeDQubit({}, {}, {})'.format(self.x, self.y, self.z)
+
+    def __str__(self):
+        return '({}, {}, {})'.format(self.x, self.y, self.z)
+
+    def _json_dict_(self):
+        return cirq.protocols.obj_to_dict_helper(self, ['x', 'y', 'z'])
+
+    def __add__(self, other: Tuple[float, float, float]) -> 'ThreeDQubit':
+        if not (isinstance(other, tuple) and len(other) == 3 and
+                all(isinstance(x, (float, int)) for x in other)):
+            raise TypeError(
+                'Can only add tuples of length 3. Was {}'.format(other))
+        return ThreeDQubit(x=self.x + other[0],
+                           y=self.y + other[1],
+                           z=self.z + other[2])
+
+    def __sub__(self, other: Tuple[float, float, float]) -> 'ThreeDQubit':
+        if not (isinstance(other, tuple) and len(other) == 3 and
+                all(isinstance(x, (float, int)) for x in other)):
+            raise TypeError(
+                'Can only subtract tuples of length 3. Was {}'.format(other))
+        return ThreeDQubit(x=self.x - other[0],
+                           y=self.y - other[1],
+                           z=self.z - other[2])
+
+    def __radd__(self, other: Tuple[float, float, float]) -> 'ThreeDQubit':
+        return self + other
+
+    def __rsub__(self, other: Tuple[float, float, float]) -> 'ThreeDQubit':
+        return -self + other
+
+    def __neg__(self) -> 'ThreeDQubit':
+        return ThreeDQubit(x=-self.x, y=-self.y, z=-self.z)
+
+
+class TwoDQubit(ThreeDQubit):
+    """A qubit in 2d."""
```

Comment:
I am a little bit confused about this at a first read, let's take this apart a bit - why does a 2D qubit inherits from a 3D qubit? Also GridQubit _is_ a 2D qubit... It would be great to clarify these design choices.

Reply:
So, the rationale behind this is that 3D qubit is the most general qubit in our devices, which can be placed in arbitrary positions in 3D. The 2D qubit is then a particular case of a 3D qubit in which the z-coordinate is fixed to 0. Some devices will only allow a 2D layout, so it's helpful to have a specific object for these devices. It also allows us to incorporate convenient static methods to create layouts in specific 2D patterns (e.g. in a triangular lattice), which would be ambivalent to have on ThreeDQubit.

As for the GridQubit, it is true that it is a 2D qubit, but it restricts the qubit positions to a grid, while we accept arbitrary positions on a plane in TwoDQubit. We are accepting GridQubit and LineQubit to give the users more flexibility, but in fact they could be completely substituted by the TwoDQubit.

Reply:
This is great, makes sense now, thank you! 
It would be very beneficial for users to be able to see this information in the pydoc for the class, so that they can more easily navigate the different options.